nrfutil/device/list.ts

```diff
@@ -46,24 +46,36 @@ export default async (
     }
 
     const onData = (data: HotplugEvent | ListEvent) => {
+        // general check I write always. Keep if it makes sense.
+        // Safeguard if undefined is passed so accessing any object property won't break with exception
+        // same can be applied to the functions above
+        if (!data) return;
```

Comment:
I generally recommend not to do that everywhere but only on the border of your system and on few important parts within. 

And then usually make it clear through the argument types that it handles undefined. 

And usually check explicitly for `null`/`undefined`, so e.g. falsy values like empty strings do not lead to surprising behaviour.

So, in short: Do not add a `if (!data) return;` here or in most other places.

Reply:
"only on the border of your system and on few important parts within" - definitely makes sense. This proposal has same motivation as one of the checks above ("Motivation behind...")